[1.21.1: NeoForge] Crash on startup with Twilight Forest

Hi! 👋 

TL;DR: 
`Exception message: net.minecraft.ResourceLocationException: Non [a-z0-9/._-] character in path of location: smallships:twilightforest:canopy_cog`

SmallShips v`2.0.0-b1.5 `

https://gist.github.com/NielsPilgaard/ae32d7eeb378c1669f9e0e2d207b8022

Similar issue with The Aether (in this case `smallships:aether:skyroot_cog`)
Seems like SmallShip tries to generate Ships for the new wood types

`java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'smallships' at 'com.talhanation.smallships.fabric.SmallshipsModFabric'!
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:403)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:401)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at knot//net.minecraft.class_310.<init>(class_310.java:477)
	at knot//net.minecraft.client.main.Main.main(Main.java:239)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.ExceptionInInitializerError
	at knot//com.talhanation.smallships.world.item.ModItems.getItem(ModItems.java)
	at knot//com.talhanation.smallships.world.item.ModItems.<clinit>(ModItems.java:16)
	at knot//com.talhanation.smallships.fabric.SmallshipsModFabric.onInitialize(SmallshipsModFabric.java:23)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:399)
	... 6 more
Caused by: net.minecraft.class_151: Non [a-z0-9/._-] character in path of location: smallships:aether:skyroot_cog
	at knot//net.minecraft.class_2960.method_45137(class_2960.java:285)
	at knot//net.minecraft.class_2960.method_60936(class_2960.java:48)
	at knot//net.minecraft.class_2960.method_60655(class_2960.java:52)
	at knot//com.talhanation.smallships.world.item.fabric.ModItemsImpl.register(ModItemsImpl.java:89)
	at knot//com.talhanation.smallships.world.item.fabric.ModItemsImpl.<clinit>(ModItemsImpl.java:80)
	... 10 more`

--- a/common/src/main/java/com/talhanation/smallships/world/item/ModItems.java
+++ b/common/src/main/java/com/talhanation/smallships/world/item/ModItems.java
@@ -26,10 +26,11 @@ public class ModItems {
     static {
         Boat.Type[] boatTypes = Boat.Type.values();
         for (Boat.Type type : boatTypes) {
-            COG_ITEMS.put(type, getItem(type.getName() + "_" + CogEntity.ID));
-            BRIGG_ITEMS.put(type, getItem(type.getName() + "_" + BriggEntity.ID));
-            GALLEY_ITEMS.put(type, getItem(type.getName() + "_" + GalleyEntity.ID));
-            DRAKKAR_ITEMS.put(type, getItem(type.getName() + "_" + DrakkarEntity.ID));
+            String name = type.getName().replaceAll("[^a-z0-9_.-]", "_");
+            COG_ITEMS.put(type, getItem(name + "_" + CogEntity.ID));
+            BRIGG_ITEMS.put(type, getItem(name + "_" + BriggEntity.ID));
+            GALLEY_ITEMS.put(type, getItem(name + "_" + GalleyEntity.ID));
+            DRAKKAR_ITEMS.put(type, getItem(name + "_" + DrakkarEntity.ID));
         }
     }
 

--- a/fabric/src/main/java/com/talhanation/smallships/world/item/fabric/ModItemsImpl.java
+++ b/fabric/src/main/java/com/talhanation/smallships/world/item/fabric/ModItemsImpl.java
@@ -76,11 +76,11 @@ public class ModItemsImpl {
         register("cannon_ball", new CannonBallItem((new Item.Properties()).stacksTo(16)));
 
         for (Boat.Type type: Boat.Type.values()) {
-
-            register(type.getName() + "_" + CogEntity.ID,  new CogItem(type, new Item.Properties().stacksTo(1)));
-            register(type.getName() + "_" + BriggEntity.ID,  new BriggItem(type, new Item.Properties().stacksTo(1)));
-            register(type.getName() + "_" + GalleyEntity.ID,  new GalleyItem(type, new Item.Properties().stacksTo(1)));
-			register(type.getName() + "_" + DrakkarEntity.ID,  new DrakkarItem(type, new Item.Properties().stacksTo(1)));
+            String name = type.getName().replaceAll("[^a-z0-9_.-]", "_");
+            register(name + "_" + CogEntity.ID,  new CogItem(type, new Item.Properties().stacksTo(1)));
+            register(name + "_" + BriggEntity.ID,  new BriggItem(type, new Item.Properties().stacksTo(1)));
+            register(name + "_" + GalleyEntity.ID,  new GalleyItem(type, new Item.Properties().stacksTo(1)));
+			register(name + "_" + DrakkarEntity.ID,  new DrakkarItem(type, new Item.Properties().stacksTo(1)));
 
         }
     }

--- a/forge/src/main/java/com/talhanation/smallships/world/item/forge/ModItemsImpl.java
+++ b/forge/src/main/java/com/talhanation/smallships/world/item/forge/ModItemsImpl.java
@@ -42,10 +42,11 @@ public class ModItemsImpl {
         register("cannon_ball", () -> new CannonBallItem((new Item.Properties()).stacksTo(16)));
 
         for (Boat.Type type: Boat.Type.values()) {
-            register(type.getName() + "_" + CogEntity.ID,  () -> new CogItem(type, new Item.Properties().stacksTo(1)));
-            register(type.getName() + "_" + BriggEntity.ID,  () -> new BriggItem(type, new Item.Properties().stacksTo(1)));
-            register(type.getName() + "_" + GalleyEntity.ID,  () -> new GalleyItem(type, new Item.Properties().stacksTo(1)));
-            register(type.getName() + "_" + DrakkarEntity.ID,  () -> new DrakkarItem(type, new Item.Properties().stacksTo(1)));
+            String name = type.getName().replaceAll("[^a-z0-9_.-]", "_");
+            register(name + "_" + CogEntity.ID,  () -> new CogItem(type, new Item.Properties().stacksTo(1)));
+            register(name + "_" + BriggEntity.ID,  () -> new BriggItem(type, new Item.Properties().stacksTo(1)));
+            register(name + "_" + GalleyEntity.ID,  () -> new GalleyItem(type, new Item.Properties().stacksTo(1)));
+            register(name + "_" + DrakkarEntity.ID,  () -> new DrakkarItem(type, new Item.Properties().stacksTo(1)));
         }
     }
 

--- a/neoforge/src/main/java/com/talhanation/smallships/world/item/neoforge/ModItemsImpl.java
+++ b/neoforge/src/main/java/com/talhanation/smallships/world/item/neoforge/ModItemsImpl.java
@@ -41,10 +41,11 @@ public class ModItemsImpl {
         register("cannon_ball", () -> new CannonBallItem((new Item.Properties()).stacksTo(16)));
 
         for (Boat.Type type: Boat.Type.values()) {
-            register(type.getName() + "_" + CogEntity.ID,  () -> new CogItem(type, new Item.Properties().stacksTo(1)));
-            register(type.getName() + "_" + BriggEntity.ID,  () -> new BriggItem(type, new Item.Properties().stacksTo(1)));
-            register(type.getName() + "_" + GalleyEntity.ID,  () -> new GalleyItem(type, new Item.Properties().stacksTo(1)));
-            register(type.getName() + "_" + DrakkarEntity.ID,  () -> new DrakkarItem(type, new Item.Properties().stacksTo(1)));
+            String name = type.getName().replaceAll("[^a-z0-9_.-]", "_");
+            register(name + "_" + CogEntity.ID,  () -> new CogItem(type, new Item.Properties().stacksTo(1)));
+            register(name + "_" + BriggEntity.ID,  () -> new BriggItem(type, new Item.Properties().stacksTo(1)));
+            register(name + "_" + GalleyEntity.ID,  () -> new GalleyItem(type, new Item.Properties().stacksTo(1)));
+            register(name + "_" + DrakkarEntity.ID,  () -> new DrakkarItem(type, new Item.Properties().stacksTo(1)));
         }
     }
 

